libpermian/plugins/anaconda_webui/__init__.py

```diff
@@ -0,0 +1,443 @@
+
+        else:
+            # Test in anaconda repo
+            if self.use_container:
+                cont_cwd = self.anaconda_dir.replace(self.temp_dir, '/root/workdir')
+                process = self._run_in_container(cmd, self.temp_dir, cwd=cont_cwd)
+            else:
+                process = subprocess.run(cmd, cwd=self.anaconda_dir, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+
+        self.log(process.stdout.decode(), 'output')
+        self.log('Test finished', show=True)
+
+        if process.returncode != 0:
+            self.reportResult(Result('complete', 'FAIL', True))
+        else:
+            self.reportResult(Result('complete', 'PASS', True))            
+
+    def teardown(self):
+        self.log('Teardown', show=True)
+        # Kill VM        
+        self._virsh_call(['destroy', self.vm_name])
+        # Remove VM
+        self._virsh_call(['undefine', self.vm_name, '--remove-all-storage'], check=True)
+        
+        self.vm_semaphore.release()
+
+        with self.setup_lock:
+            # If this instance is the last one using temp dir -> remove it
+            self.instances[self.git_anaconda_branch].remove(self)
+            if len(self.instances[self.git_anaconda_branch]) == 0:
+                self.temp_dirs[self.git_anaconda_branch].cleanup()
+                
+                del self.instances[self.git_anaconda_branch]
+                # Remove container image, if this was the last anacoda-webui workflow
+                if self.instances == dict():
+                    self.container.remove_image()
+
+    def terminate(self):
+        # Kill VM
+        self._virsh_call(['destroy', self.vm_name])
+
+    def log(self, message, name='workflow', show=False):
+        if name == 'workflow' and show:
+            self.last_log = message
+        super().log(message, name)
+
+    def displayStatus(self):
+        return self.last_log
+
+    def _start_vm(self):
+        """Prepare virt-install command, start VM and wait for IP address
+
+        :return: IP address
+        :rtype: str
+        """
+        self.log('Starting VM', show=True)
+
+        # Get compose location
+        os_url = self.installation_source.repos[self.installation_source.base_repo][self.architecture]
+        if self.installation_source.kernel_path and self.installation_source.initrd_path:
+            location = f'{os_url},kernel={self.installation_source.kernel_path},initrd={self.installation_source.initrd_path}'
+        else:
+            location = os_url
+
+        # Construct kernel cmdline for --extra-args
+        kernel_cmdline = f'inst.sshd inst.webui inst.webui.remote inst.stage2={self.os_url}'
+        if self.additional_repos:
+            for repo in self.additional_repos:
+                kernel_cmdline += f' inst.addrepo={self.installation_source.repo[repo][self.architecture]}'
```

Comment:
I have 2 notes here:

1. The repoid is missing. The inst.addrepo requires the argument to be in form: `${repoid}.${url}`
2. Since the testcase automation data field is named `additional_repos`, I expected the repos to be URLs, not ids, so that one could be able to provide custom URL (e.g. with additional payload content), but If I'm not mistaken, this is not what this code does. Yes, the kernel_cmdline can be used for that, it's just that the current naming/use is misleading.

Reply:
The code should work now. Is there a plan to introduce a different way how to specify repositories? Something like `custom_repos` where one would provide both ID and url of the repos they would like to add.

Reply:
There is no need for it right now, it should be easy to add it later.